### DISCUSSION TITLE
docx to markdown, how to ignore certain things (like tables and images)

### DISCUSSION THREAD

**Discussion by michellepace** [jgm/pandoc#9935] [Q&A] [answered]
Hello there. 

Using Python in a Colab Notebook, I need to convert a Word into markdown so that I can send it to a large language model in chunks. The demo of Pandas looks brilliant! I've tried doing this myself by using panda-docx to read the docx and then writing the below function. I am so happy I found Pandoc and hope I can entirely replace panda-docx and my function with it.

**My Problem:** 
- I only want to write **Paragraphs**, **headings** (a style of paragraph I believe), and **Lists** (another style of paragraph I think) to my markdown file (maybe **Tables** but I'm not yet sure). 
- I would like to ignore **images**, **table of contents**, **footers**, **headers**, **titles** etc. Is it possible to do this and if so could you please share a reference url with me?
![image](https://github.com/jgm/pandoc/assets/512813/3bf08679-889f-4a47-9a3a-b535083e1c37)

Thanks in advance and can't wait for a reply!
Michelle
![image](https://github.com/jgm/pandoc/assets/512813/3a43edab-1fd7-4409-a2e2-576c0c2fc618)

**Comment by tarleb** [comment, marked as answer]
This can be done with a pandoc [filter](https://pandoc.org/filters). Below is a [Lua filter](https://pandoc.org/lua-filters), which should work out-of-the-box, without having to install additional software. However, if you prefer a more pythonic solution, then checkout "panflute".

``` lua
local function omit ()
  return {}
end

return {{
  Table = omit,
  Figure = omit,
  Image = omit,
  -- and so on.
}}
```

Save the above to a file and pass it to pandoc via the `--lua-filter` command line parameter.

**Reply by michellepace** [reply]
Thank you very much @tarleb , pandoc is lovey. Unfortunately though I need this to run in a Colab notebook and I didn’t realise I would have to install pandoc on my machine. But thank you very much for your reply. I think I’m stuck with the python-docx library.
